TeamCity.ServiceMessages/Write/Special/ITeamCityProgressWriter.cs

```diff
@@ -0,0 +1,41 @@
+﻿namespace JetBrains.TeamCity.ServiceMessages.Write.Special
+{
+    using System;
+
+    /// <summary>
+    /// <para>
+    /// Use <see cref="WriteProgress"/> method to add progress message
+    /// <pre>
+    ///     ##teamcity[progressMessage '&lt;message text>' ]
+    /// </pre>
+    /// </para>
+    ///
+    /// <para>
+    /// Use <see cref="OpenProgress"/> method to add progress message
+    /// <pre>##teamcity[progressStart '&lt;message>']</pre>
+    /// and
+    /// <pre>##teamcity[progressFinish '&lt;message>']</pre>
+    /// on writer dispose.
+    /// </para>
+    /// <para>
+    /// http://confluence.jetbrains.net/display/TCD18/Build+Script+Interaction+with+TeamCity#BuildScriptInteractionwithTeamCity-ReportingMessagesForBuildLog
+    /// </para>
+    /// </summary>
```

Comment:
A couple of remarks:
* Some minor grammatical changes will make the sentences sound better, e.g. "Use **the** <see cref="WriteProgress"/> method to add **a** progress message", "Use **the** <see cref="OpenProgress"/> method to add **a** progress message".
* For documentation, we should use a link to the latest version on https://www.jetbrains.com/help: https://www.jetbrains.com/help/teamcity/service-messages.html#Reporting+Build+Progress

I suggest something like this:
```suggestion
    /// <summary>
    /// <para>
    /// Use the <see cref="WriteProgress"/> method to add a progress message:
    /// <pre>##teamcity[progressMessage '&lt;message text>' ]</pre>
    /// </para>
    ///
    /// <para>
    /// Use the <see cref="OpenProgress"/> method to add a progress start message:
    /// <pre>##teamcity[progressStart '&lt;message>']</pre>
    /// and a
    /// <pre>##teamcity[progressFinish '&lt;message>']</pre>
    /// progress finish message on writer dispose.
    /// </para>
    /// <para>
    /// https://www.jetbrains.com/help/teamcity/service-messages.html#Reporting+Build+Progress
    /// </para>
    /// </summary>
```

Reply:
Sorry, the documentation link in my previous comment was wrong. I have edited the comment to include the correct link.